src/utils/documentdb.ts

```diff
@@ -790,4 +791,31 @@ export function queryAttachments<T>(
   };
 }
 
+/**
+ * A convenient type that extends Cosmos' DocumentDb.QueryError with a generic status that maps a runtime error (instance of Error)
+ */
+export type QueryError =
+  | DocumentDb.QueryError
+  | { code: "error"; body: string };
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ *
+ * @returns either the query result or a query failure
+ */
+export const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<DocumentDb.QueryError | Error, R>>
```

Comment:
Why the signature of this promise is 

`Promise<Either<DocumentDb.QueryError | Error, R>>` and not just

`Promise<Either<DocumentDb.QueryError, R>>` ?

Reply:
For a more relaxed API, but I think we can be more specific